lib/CGameState.cpp

```diff
@@ -2133,6 +2133,73 @@ int3 CGameState::guardingCreaturePosition (int3 pos) const
 	return gs->map->guardingCreaturePositions[pos.x][pos.y][pos.z];
 }
 
+void CGameState::updateRumor()
+{
+	static std::vector<RumorState::ERumorType> rumorTypes = {RumorState::RUMOR_MAP, RumorState::RUMOR_STATS, RumorState::RUMOR_RAND, RumorState::RUMOR_RAND};
+	static std::vector<int> statsRumorTypes = {208, 209, 210, 211, 212};
```

Comment:
Why are these magic numbers needed? What about one more enum? If you need these constants then just define enum with fixed values ( RUMOR_GRAIL = 212 )